packages/components/pages/f-mfa/src/services/EventBuilder.js

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable camelcase */
+// eslint-disable-next-line import/prefer-default-export
+export const buildEvent = (key, data = '') => {
+    switch (key) {
+        case 's1b':
+            return {
+                EventName: 'form_view',
+                Parameters: {
+                    component_name: 'multiFactorAuthentication',
+                    component_type: 'form',
+                    form_action: 'view'
+                }
+            };
+        case 's2':
+            return {
+                EventName: 'dialog_view',
+                Parameters: {
+                    component_name: 'multiFactorAuthenticationbadActor',
+                    component_type: 'dialog',
+                    dialog_action: 'view',
+                    dialog_type: 'info'
+                }
+            };
+        case 's3':
+            return {
+                EventName: 'dialog_select',
+                Parameters: {
+                    component_name: 'multiFactorAuthenticationbadActor',
+                    component_type: 'dialog',
+                    dialog_action: 'select',
+                    dialog_type: 'info',
+                    dialog_buttonText: 'go back'
+                }
+            };
+        case 's4':
+            return {
+                EventName: 'form_error',
+                Parameters: {
+                    component_name: 'multiFactorAuthentication',
+                    component_type: 'form',
+                    form_action: 'error',
+                    error_message: `${data}`
```

Comment:
here you can just do `data`?

Reply:
The only difference is that this does force it to be a string, even if someone were to pass an object in. TypeScript would save us here!